Выполнил первый проект

Artem-Practic opened the pull request:
Отправляю на ревью

Artem-Practic committed:
Выполнил первый проект

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+  white-space: nowrap;
+}
+
+.intro__main-text p {
```
https://disk.yandex.com/i/d-NojUh0fGu1NQ

Некорректно указан межстрочный интервал. 
Это возникает из-за того, что было взято некорректное значение из макета.
Сверьте значение в верстке со значением в макете.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+  letter-spacing: normal;
+}
+
+.content-section__introduction {
```
https://disk.yandex.com/i/ZnJqHfK_7JtuvQ

Некорректно указан межстрочный интервал. 
Это возникает из-за того, что было взято некорректное значение из макета.
Сверьте значение в верстке со значением в макете.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+  row-gap: 16px;
+}
+
+.content-section__list-accent {
```
https://disk.yandex.com/i/Pri5NwK_At9tvw

Некорректно указан отступ. 
Это возникает из-за того, что было взято некорректное значение из макета.
Сверьте значение в верстке со значением в макете.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+              Вот основные:
+            </p>
+            <ul class="content-section__list">
+              <li class="content-section__list-item"><span class="content-section__list-accent">Неуверенность в результате.</span> И в том, что его вообще можно достичь. Ведь как можно быть уверенным, что проводишь всю работу не впустую? Что доведёшь дело до конца? А тут, пока не начал, не надо и заканчивать.</li>
```
Некорректно указан межстрочный интервал. 
Это возникает из-за того, что было взято некорректное значение из макета.
Сверьте значение в верстке со значением в макете.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+              <div class="card__content">
+                <h3 class="card__title">Начать любыми словами</h3>
+                <div class="card__text">
+                  <p>
```
Некорректно указан межстрочный интервал. 
Это возникает из-за того, что было взято некорректное значение из макета.
Сверьте значение в верстке со значением в макете.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+  font-weight: bold;
+}
+
+.how-to {
```
https://disk.yandex.com/i/i7MZfQ6SE1P1xA

Некорректно указан отступ. 
Это возникает из-за того, что было взято некорректное значение из макета.
Сверьте значение в верстке со значением в макете.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+      <a class="header__logo">
+        <img class="header__logo-image" src="images/logo.svg" alt="логотип вёрстка">
+      </a>
+      <nav class="header__menu">
```
• Не забыли выделить навигацию при помощи `nav`.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+        <img class="header__logo-image" src="images/logo.svg" alt="логотип вёрстка">
+      </a>
+      <nav class="header__menu">
+        <ul class="header__menu-list">
```
• Отличное решение воспользоваться списком для перечисляемых элементов.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+      </a>
+      <nav class="header__menu">
+        <ul class="header__menu-list">
+          <li class="header__menu-item header__menu-item_active">
```
• Хорошо применили блок/элемент и его модификатор.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+      </section>
+      <section class="intro content-section content-section_theme_dark">
+        <header class="intro__heading">
+          <h3 class="intro__title">8 способов борьбы со страхом чистого листа</h3>
```
Здесь следует использовать заголовок более высокого уровня. 
Это соответствует требованиям макета и размеру шрифта. 
Также обратите внимание на уровни заголовков, используемые выше в коде, в других блоках. Вы можете обратиться к документации по элементам заголовков для получения дополнительной информации:
[Уровни заголовков](https://developer.mozilla.org/ru/docs/Web/HTML/Element/Heading_Elements)

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+      <section class="intro content-section content-section_theme_dark">
+        <header class="intro__heading">
+          <h3 class="intro__title">8 способов борьбы со страхом чистого листа</h3>
+          <time class="intro__date" datetime="2022-06-05">05.06.2022 г.</time>
```
• Подходящий тег, чтобы выделить временные рамки или дату.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+          <time class="intro__date" datetime="2022-06-05">05.06.2022 г.</time>
+        </header>
+        <div class="intro__main-text">
+          <p>Начинать что-то новое — тяжело и страшно.
```
• Считается  хорошей практикой использовать параграф для выделения большого или некоторого количества текстовой информации.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+        </div>
+      </section>
+      <section class="reasons content-section content-section_theme_bright">
+        <h2 class="content-section__subtitle"><span class="accent-span text-right">Почему чистый лист</span><span>— такой cтрашный?</span></h2>
```
Здесь следует использовать заголовок более меньшего уровня. 
Это соответствует требованиям макета и размеру шрифта. 
Также обратите внимание на уровни заголовков, используемые выше в коде, в других блоках. Вы можете обратиться к документации по элементам заголовков для получения дополнительной информации:
[Уровни заголовков](https://developer.mozilla.org/ru/docs/Web/HTML/Element/Heading_Elements)

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+      </section>
+      <section class="reasons content-section content-section_theme_bright">
+        <h2 class="content-section__subtitle"><span class="accent-span text-right">Почему чистый лист</span><span>— такой cтрашный?</span></h2>
+        <p class="content-section__introduction">Сам по себе чистый лист, конечно, не страшный.
```
* Необходимо отформатировать код в `html`: не соблюдены отступы  по отношению к другим блокам/элементам.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+             Причин, почему первые усилия даются так сложно, может быть очень много.
+              Вот основные:
+            </p>
+            <ul class="content-section__list">
```
• Отличное решение воспользоваться списком для перечисляемых элементов.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+            </ul>
+      </section>
+      <section class="how-to content-section content-section_theme_dark">
+        <h2 class="content-section__subtitle"><span class="accent-span text-right">Как всё-таки начать</span><span>что-то делать</span></h2>
```
Здесь следует использовать заголовок более меньшего  уровня. 
Это соответствует требованиям макета и размеру шрифта. 
Также обратите внимание на уровни заголовков, используемые выше в коде, в других блоках. Вы можете обратиться к документации по элементам заголовков для получения дополнительной информации:
[Уровни заголовков](https://developer.mozilla.org/ru/docs/Web/HTML/Element/Heading_Elements)

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+            чтобы переключить мозг из состояния паники в состояние потока.
+             Причин, почему первые усилия даются так сложно, может быть очень много.
+              Вот основные:
+            </p>
```
• Считается  хорошей практикой использовать параграф для выделения большого или некоторого количества текстовой информации.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+        <ul class="cards-list">
+          <li>
+            <article class="card card_theme_dark">
+              <div class="card__author">
```
**Можно лучше**:
Для более точной и правильной семантической разметки документов в HTML5 были добавлены новые теги, такие как `figure` и `figcaption`, которые могут использоваться для изображений и их описаний. 
Тег `figure` используется для группировки элементов, связанных с изображением, например, с его описанием или подписью. Тег  `figcaption` используется для указания подписи к изображению внутри контейнера `figure`. Это позволяет улучшить семантику разметки, а также повысить доступность контента для пользователей с ограниченными возможностями.
https://developer.mozilla.org/ru/docs/Web/HTML/Element/figure

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+        </p>
+        <ul class="cards-list">
+          <li>
+            <article class="card card_theme_dark">
```
• Выбрали подходящий тег `article` для контентной информации, которая потом может быть использована снова в другой части сайта.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+                <p class="card__author-name">Алексей Поляринов</p>
+              </div>
+              <div class="card__content">
+                <h3 class="card__title">Начать любыми словами</h3>
```
Здесь следует использовать заголовок более меньшего уровня. 
Это соответствует требованиям макета и размеру шрифта. 
Также обратите внимание на уровни заголовков, используемые выше в коде, в других блоках. Вы можете обратиться к документации по элементам заголовков для получения дополнительной информации:
[Уровни заголовков](https://developer.mozilla.org/ru/docs/Web/HTML/Element/Heading_Elements)

vgschevchenko replied:
Здесь следует использовать заголовок более меньшего  уровня. 
Это соответствует требованиям макета и размеру шрифта. 
Также обратите внимание на уровни заголовков, используемые выше в коде, в других блоках. Вы можете обратиться к документации по элементам заголовков для получения дополнительной информации:
[Уровни заголовков](https://developer.mozilla.org/ru/docs/Web/HTML/Element/Heading_Elements)

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+                <div class="card__text">
+                  <p>
+                    Любыми — без шуток. Если не верите нам, посмотрите, например, чем открывается эссе Алексея Поляринова «Чарли Кауфман: расширение пространства». Можно и менее радикально. Главное, чтобы пустую страницу начали заполнять символы. Получится своеобразная ловушка для мозга: он видит, что вы уже начали что-то делать, и ему становится легче войти в состояние потока.
+                  </p>
```
• Считается  хорошей практикой использовать параграф для выделения большого или некоторого количества текстовой информации.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+      </section>
+    </main>
+    <footer class="footer content-section content-section_theme_dark">
+      <div class="footer__copy"><p>© VЁRSTKA</p></div>
```
Текстовые блоки необходимо размечать тегами `<p>` (параграф). 
Пожалуйста, исправьте разметку текстовых блоков, используя теги `<p>`. Это семантически правильный подход, который поможет правильно структурировать контент и улучшит его доступность для поисковых систем и пользователей.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,136 @@
+    </main>
+    <footer class="footer content-section content-section_theme_dark">
+      <div class="footer__copy"><p>© VЁRSTKA</p></div>
+      <nav class="footer__links">
```
Это не навигация, поэтому тег `nav` тут не очень подходит.
Тут просто список ссылок с иконками внутри. 
Как решение вы можете превратить этот элемент `nav` в список, а логотипы сделать элементами этого списка.

vgschevchenko replied:
Это не навигация, поэтому тег nav тут не очень подходит.
Тут просто список ссылок с иконками внутри.
Как решение вы можете превратить этот элемент nav в список, а логотипы сделать элементами этого списка.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+  border-bottom: 1px solid #f5f5f5;
+}
+
+.header__menu-item_active a {
```
Хорошее решение применить вложенность селекторов.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+}
+
+.cover__title {
+  font-family: "EB Garamond";
```
**Комментарий к исправлению:**
Для обеспечения более надежного отображения текста на сайте, помимо основного шрифта, рекомендуется указать альтернативные варианты шрифтов. Это обеспечит поддержку в случае, если основные шрифты не загрузятся или не будут поддержаны браузером. Например, можно указать системные шрифты в качестве запасных, такие как `Arial, sans-serif`. Это гарантирует более надежное отображение контента на различных устройствах и в разных условиях.

### Пожалуйста, пройдитесь по всему файлу/файлам и исправьте данную ошибку везде, где она присутствует. Работа будет отправлена на доработку до тех пор, пока все места с этой ошибкой не будут исправлены или пока не будет проведена полная проверка файлов на наличие данной ошибки.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+}
+
+.content-section__subtitle {
+  font-family: "EB Garamond";
```
**Комментарий к исправлению:**
Для обеспечения более надежного отображения текста на сайте, помимо основного шрифта, рекомендуется указать альтернативные варианты шрифтов. Это обеспечит поддержку в случае, если основные шрифты не загрузятся или не будут поддержаны браузером. Например, можно указать системные шрифты в качестве запасных, такие как `Arial, sans-serif`. Это гарантирует более надежное отображение контента на различных устройствах и в разных условиях.

### Пожалуйста, пройдитесь по всему файлу/файлам и исправьте данную ошибку везде, где она присутствует. Работа будет отправлена на доработку до тех пор, пока все места с этой ошибкой не будут исправлены или пока не будет проведена полная проверка файлов на наличие данной ошибки.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+}
+
+.card__title {
+  font-family: "EB Garamond";
```
**Комментарий к исправлению:**
Для обеспечения более надежного отображения текста на сайте, помимо основного шрифта, рекомендуется указать альтернативные варианты шрифтов. Это обеспечит поддержку в случае, если основные шрифты не загрузятся или не будут поддержаны браузером. Например, можно указать системные шрифты в качестве запасных, такие как `Arial, sans-serif`. Это гарантирует более надежное отображение контента на различных устройствах и в разных условиях.

### Пожалуйста, пройдитесь по всему файлу/файлам и исправьте данную ошибку везде, где она присутствует. Работа будет отправлена на доработку до тех пор, пока все места с этой ошибкой не будут исправлены или пока не будет проведена полная проверка файлов на наличие данной ошибки.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+}
+
+.footer__links {
+  width: 73px;
```
Использование фиксированной высоты/ширины для текстовых блоков может привести к проблемам с масштабируемостью. Например, если количество текста окажется больше чем текущая ширина/высота элемента сможет вместить, то верстка просто сломается.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,244 @@
+
+.footer__links {
+  width: 73px;
+  height: 18px;
```
Использование фиксированной высоты/ширины для текстовых блоков может привести к проблемам с масштабируемостью. Например, если количество текста окажется больше чем текущая ширина/высота элемента сможет вместить, то верстка просто сломается.

vgschevchenko reviewed (commented):
#### Здравствуйте, Артем. 

#### Нужно развернуть общий комментарий, что бы увидеть его полностью (⬇👇)

##### Это общий комментарий. Большая часть ошибок, а также их объяснение и конкретные места указаны в самом коде. 

Хорошая работа, но еще осталось над чем потрудиться.
 
- Ваши усилия и преодоление трудностей не остаются без внимания - продолжайте в том же духе!

Исправьте все ошибки и работа будет принята. 
Вы справитесь, желаю вам успехов. 

### Что понравилось в работе:

* В консоли нет ошибок
* Верстка выполнена аккуратно.
* Изображения не теряют своих пропорций и не искажаются. 
* Код в целом хорошо отформатирован.
* Хорошее совпадение с брифом/макетом.
* Постарались не дублировать код.
* Используется семантические теги: `main, h, p, ul, nav, button, article, time, section, header, footer` и т.д.
* Постарались использовать названия классов в соответствии с рекомендациями из задания.
* Постарались использовать названия классов в соответствии с БЭМ методологией.
* Валидатор `html/css` не ругается на ошибки. 
* Элементы, которые относятся к декору - выполнены соответствующим образом при помощи фона, `div, span` или псевдоэлементов. 
* Альтернативный текст для изображений заполнен корректно и на языке страницы. 
* Для большинства контентных или текстовых элементов и блоков не используется фиксированная высота /ширина. 
* Для построения элементов использовали flex/grid. 






### Но есть так же и замечания, которые потребуют доработки. 

### Места, которые надо исправить в проекте: 

* Необходимо свериться с макетом и не забыть указать корректные размеры или отступы между блоками или их элементами.  
 * Обратите внимание на отступы и межстрочный интервал у текстовых элементов. 
* Обратите внимание на уровни заголовков. Важно придерживаться иерархии заголовков, чтобы обеспечить правильную семантику страницы. Необходимо следовать порядку и использовать заголовки от h1 до h6 по их значимости и вложенности.
* Отформатировать код `html` в едином стиле с соблюдением отступов и т.д. 
	* Код в `html` должен быть похож на лесенку и таким образом будет видна вложенность одних  блоков/элементов внутри других. 
	* Если вы оставляете два пробела/таба для отступов, то необходимо придерживается такого подхода на протяжении всего кода. Исправьте в других файлах или участках кода. 
	* **Используйте `Prettier` для автоформатирования кода.**
* Текстовые блоки необходимо размечать тегами <p> (параграф).
* Воспользоваться семантическими тегами, там где это действительно необходимо, например: `nav, ul, address, article, header, main, section, footer` и т.д.
* Для обеспечения более надежного отображения текста на сайте, помимо основного шрифта, рекомендуется указать альтернативные варианты шрифтов. 
* Использование фиксированной высоты/ширины/грид-колонки для текстовых блоков может привести к проблемам с масштабируемостью. 




# Что можно улучшить:

* Для более точной и правильной семантической разметки документов в HTML5 были добавлены новые теги, такие как figure и figcaption, которые могут использоваться для изображений и их описаний.

Комментарии *"Можно лучше"* носят рекомендательный характер, их выполнение остается на Ваше усмотрение.

Вносите правки без спешки, следите за тем, что изменилось после внесения той или иной правки и при этом больше ничего не сломалось. Убедитесь перед отправкой, что все правки внесены, основные требования соответствуют брифу и чек листу. Если появятся трудности - обращайтесь к наставнику и старшему студенту, они Вам помогут.

Успешной работы над проектом, и до встречи на следующем ревью!

Artem-Practic committed:
Внес правки после первого ревью - вариант 1

Artem-Practic committed:
Внес правки после первого ревью - вариант 2

Правка стилей

Artem-Practic committed:
Внес правки после первого ревью - вариант 3

Artem-Practic committed:
Внес правки после первого ревью - вариант 2

Artem-Practic committed:
Внес правки после первого ревью - вариант 3

Правка style.css
Форматирование index.html, style.css, fonts.css при помощи Prettier

Artem-Practic committed:
Результирующие правки после первого ревью

Artem-Practic committed:
Результирующие правки после первого ревью

Artem-Practic commented:
Добрый день. Внес правки после ревью.

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,222 @@
+        </div>
+      </section>
+      <section class="reasons content-section content-section_theme_bright">
+        <h2 class="content-section__subtitle">
```
Здесь следует использовать заголовок более меньшего  уровня. 
Это соответствует требованиям макета и размеру шрифта. 
Также обратите внимание на уровни заголовков, используемые выше в коде, в других блоках. Вы можете обратиться к документации по элементам заголовков для получения дополнительной информации:
[Уровни заголовков](https://developer.mozilla.org/ru/docs/Web/HTML/Element/Heading_Elements)

vgschevchenko commented on index.html:
```diff
@@ -0,0 +1,222 @@
+        </ul>
+      </section>
+      <section class="how-to content-section content-section_theme_dark">
+        <h2 class="content-section__subtitle">
```
Здесь следует использовать заголовок более меньшего  уровня. 
Это соответствует требованиям макета и размеру шрифта. 
Также обратите внимание на уровни заголовков, используемые выше в коде, в других блоках. Вы можете обратиться к документации по элементам заголовков для получения дополнительной информации:
[Уровни заголовков](https://developer.mozilla.org/ru/docs/Web/HTML/Element/Heading_Elements)

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,245 @@
+}
+
+.footer__links {
+  min-width: 73px;
```
**Можно лучше:**

Высота для данного элемента/блока будет лишней - она зависит от вложенного в него контента. 
В противном случае, когда контента станет меньше, то блок не изменит свою высоту и под ним останется пустое пространство. Мы не задаем фиксированную  или жёстко-заданную высоту для контентных или текстовых элементов. 
Если ошибка повторяются в других местах, например в медиа-запросах, то ее также необходимо исправить. 

* Можно задать внутренние отступы (padding) для блока, что бы суммарно с высотой контента блок имел требуемую высоту.

vgschevchenko commented on styles/style.css:
```diff
@@ -0,0 +1,245 @@
+
+.footer__links {
+  min-width: 73px;
+  min-height: 18px;
```
**Можно лучше:**

Высота для данного элемента/блока будет лишней - она зависит от вложенного в него контента. 
В противном случае, когда контента станет меньше, то блок не изменит свою высоту и под ним останется пустое пространство. Мы не задаем фиксированную  или жёстко-заданную высоту для контентных или текстовых элементов. 
Если ошибка повторяются в других местах, например в медиа-запросах, то ее также необходимо исправить. 

* Можно задать внутренние отступы (padding) для блока, что бы суммарно с высотой контента блок имел требуемую высоту.

vgschevchenko reviewed (approved):
## Поздравляю! Ваша работа принята!

Спасибо вам за терпение и приложенные усилия.))
Вы молодец, хорошо старались, не сдались и не опустили руки. 
Присутствует моменты, которые необходимо поправить, чтобы работа стала еще лучше и ее можно было добавить в портфолио. 

Желаю вам хороших каникул, также успехов в обучении и в освоении нового.

У Вас получился отличный хороший.
